### PR TITLE
feat: implement user management functions as SQL UDFs

### PR DESCRIPTION
**Pull request by patrickdevivo**
Exposes some user management functionality to the GraphQL API

- `mergestat.user_mgmt_pg_users` view to list all pg users
- `mergestat.user_mgmt_add_user` to add a new user with a password
- `mergestat.user_mgmt_remove_user` to drop an existing user
- `mergestat.user_mgmt_update_user_password` to update the password of a user
- `mergestat.user_mgmt_set_user_role` to set the MergeStat role a user has

### CODE REVIEW

File: migrations/900000000000003_user_management.up.sql [hunk 1]
```diff
@@ -0,0 +1,101 @@
+BEGIN;
+
+-- View to list all users in the database
+CREATE OR REPLACE VIEW mergestat.pg_users AS (
+    SELECT * FROM pg_catalog.pg_user
+);
```
Comment (patrickdevivo) [inline migrations/900000000000003_user_management.up.sql:6]: reduce (for now) to just `username` and `id`

File: migrations/900000000000004_user_management.up.sql [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+DROP FUNCTION IF EXISTS mergestat.add_user(NAME, TEXT);
+
+CREATE OR REPLACE FUNCTION mergestat.add_user(username NAME, password TEXT)
+RETURNS SMALLINT AS
+$BODY$
+DECLARE
+BEGIN
+    -- Create the user with the given password
+    EXECUTE FORMAT('CREATE USER %I WITH PASSWORD %L', username, password);
+    RETURN 1;
```
Comment (patrickdevivo) [inline migrations/900000000000004_user_management.up.sql:89]:
```sql
GRANT %I TO mergestat_admin
```

Comment (patrickdevivo) [reply migrations/900000000000004_user_management.up.sql:89]:
```sql
GRANT %I TO readaccess
```

File: migrations/900000000000004_user_management.up.sql [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+$BODY$
+DECLARE
+BEGIN
+    -- first revoke all existing mergestat roles from the user
+    EXECUTE FORMAT('REVOKE mergestat_role_readonly FROM %I', username);
+    EXECUTE FORMAT('REVOKE mergestat_role_user FROM %I', username);
+    EXECUTE FORMAT('REVOKE mergestat_role_admin FROM %I', username);
+    CASE
+        WHEN role = 'ADMIN' THEN
+            EXECUTE FORMAT('GRANT mergestat_role_admin TO %I', username);
```
Comment (patrickdevivo) [inline migrations/900000000000004_user_management.up.sql:108]:
```sql
ALTER USER nick WITH CREATEROLE;
```

File: migrations/900000000000004_user_management.up.sql [hunk 4]
```diff
@@ -0,0 +1,156 @@
+BEGIN;
```
Comment (patrickdevivo) [inline migrations/900000000000004_user_management.up.sql:1]: let's prefix all the functions/views added here with a common prefix like `user_management` for consistency/easier finding

**Review by amenowanna — APPROVED** [review]

**Pull request merged**